sycl/unittests/helpers/CommonRedefinitions.hpp

```diff
@@ -0,0 +1,136 @@
+//==---- CommonRedefinitions.hpp --- Header with common PI redefinitions ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <helpers/PiImage.hpp>
+#include <helpers/PiMock.hpp>
+
+inline pi_result redefinedProgramCreateCommon(pi_context, const void *, size_t,
+                                              pi_program *) {
+  return PI_SUCCESS;
+}
+
```

Comment:
Nit. I would allocate something like `int *` and release it just in case we test somewhere that the program is not null. This can be applied as a separate PR, though.